### PR TITLE
feat(Rest): force uppercase methods

### PR DESCRIPTION
**Pull request by doktordirk**

### CODE REVIEW

File: src/rest.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ export class Rest {
   request(method, path, body, options = {}) {
     let requestOptions = extend(true, {}, this.defaults, options);
 
-    requestOptions.method = method;
+    // Force method to be upper-case for pre-flight Accept-Methods matching
+    requestOptions.method = method.toUpperCase();
```
Comment (RWOverdijk) [inline src/rest.js]: I still disagree with this.

Comment (doktordirk) [inline src/rest.js]: ? i the original you agreed, or

Comment (RWOverdijk) [inline src/rest.js]: I didn't :p I think this should be solved server-side. The thing is, this small stupid change is a BC-break. With the new major version coming up, I'd say we can go with this if it brings happiness :D

**Comment by doktordirk** [comment]
fine by me

**Pull request closed without merging**